src/blockchain/shelley/utils.js

```diff
@@ -4,25 +4,66 @@ import type { PoolRegistrationType, PoolRetirementType, StakeDelegationType } fr
 import { CERT_TYPE } from './certificate'
 import type { ShelleyTxType } from './tx'
 import { AddressKind } from '../../../js-chain-libs/pkg/js_chain_libs';
+import { PublicKey } from 'js-chain-libs/js_chain_libs';
+
+function free(...args) {
+  for (const a of args) {
+    if (a && a.free) {
+      a.free()
+    }
+  }
+}
+
+function consumeAccountToOptionalAddress(account, discrimination, stringEncoding = 'hex'): string {
+  if (!account) {
+    return null
+  }
+  const address = account.to_address(discrimination);
+  const result = Buffer.from(address.as_bytes()).toString(stringEncoding);
+  free(address, account)
+  return result
+}
+
+function consumeKeysToStrings(keys, stringEncoding = 'hex'): Array<string> {
+  const result: Array<string> = []
+  for (let i = 0; i < keys.size(); i += 1) {
+    const key = keys.get(i);
+    const keyBytes = Buffer.from(key.as_bytes())
+    result.push(keyBytes.toString(stringEncoding))
+    free(key)
+  }
+  free(keys)
+  return result
+}
 
 // frees input rust-wasm Value and parses it into a js number
-const consumeValueToNumber = (value: any): number => {
+const consumeOptionalValueToNumber = (value: any): number => {
   // TODO: Values are returned as strings under the rationale that js strings
   // can only fit a 52-bit radix as integers, but since the max ADA supply is smaller
   // than this (but bigger than a 32-bit int) this should be safe. We should try and
   // see if this can be changed in js-chain-libs and use that there instead.
+  if (!value) {
+    return null
```

Comment:
Is the plan to just pretend that it's `number` and not `?number` as the return type? If not, then we'd be wanting to add in explicit checks on the return. Where were we getting null inputs anyway?

Reply:
Yep, incorrect return type. Value might be null in the pool-registration cert in the `limit` parameter, when `max_limit()` is queried from Rust object. That field is optional and might null.

I'm changing the return type to `?number` 👌